### PR TITLE
In game interface for creating server polls

### LINKED ISSUE #7003: Let server leads create in-game polls without needing direct database access
Feature Request

I'd like some way for server leads (Headmins, Hosts) to create an in-game poll (Like the ones you can access from the lobby menu) without needing direct database access. Preferably the method should remain in-game.

This may involve @mrstonedone so I'll ping him here.

### PR DESCRIPTION
**Pull request by Jordie0608**
Adds an interface to allow admins to make server polls without direct DB access or fiddling with SQL.
Usable by any admin with +PERMISSIONS as we can't directly check for Headmin rank.

Updates schema for 'poll_question' table to format used by our servers (thanks erro).

Resolves #7003

### CODE REVIEW

File: code/modules/admin/create_poll.dm [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	set category = "Special Verbs"
+	if(!check_rights(R_PERMISSIONS))	return
+	if(!dbcon.IsConnected())
+		src << "<span class='danger'>Failed to establish database connection.</span>"
+		return
+	var/polltype = input("Choose poll type.","Poll Type") in list("Single Option","Text Reply","Rating","Multiple Choice")
+	var/choice_amount = 0
+	switch(polltype)
+		if("Single Option")
+			polltype = "OPTION"
```
Comment (duncathan) [inline code/modules/admin/create_poll.dm]: surely these can be defines

Comment (Jordie0608) [inline code/modules/admin/create_poll.dm]: They need to be passed to SQL as text and I'd prefer to not try and find out what happens if you try and pass a define to SQL.

Comment (duncathan) [inline code/modules/admin/create_poll.dm]: ...defines are preproccessor.
`#define POLLTYPE_OPTION "OPTION"` would replace all instances of POLLTYPE_OPTION in the code with "OPTION" /before/ compiling.

Comment (Jordie0608) [inline code/modules/admin/create_poll.dm]: I'm aware, but this is BYOND and SQL here.
I'll do touchups of poll's code like this later, probably in another PR.

Comment (MrStonedOne) [inline code/modules/admin/create_poll.dm]: I will state that all other game code that deals with polls doesn't use these defines.

And defines for a static string that says the same thing and isn't changing and for sql reasons can't change to a number just doesn't make sense.

**Comment by Dorsisdwarf** [comment]
There's an issue this fixes, hang on

File: code/modules/admin/create_poll.dm (hunk 1 above)
Comment (Miauw62) [inline code/modules/admin/create_poll.dm]: would you rather have code compile and not work than have code not compile at all?

File: SQL/tgstation_schema.sql [hunk 2]
```diff
@@ -263,6 +263,9 @@ CREATE TABLE `poll_question` (
   `question` varchar(255) NOT NULL,
   `adminonly` tinyint(1) DEFAULT '0',
   `multiplechoiceoptions` int(2) DEFAULT NULL,
+  `createdby_ckey` varchar(45) NULL DEFAULT NULL,
+  `createdby_ip` varchar(45) NULL DEFAULT NULL,
+  `for_trialmin` varchar(45) NULL DEFAULT NULL,
```
Comment (Markolie) [inline SQL/tgstation_schema.sql]: Out of curiosity: why add this column? I don't see it being used anywhere in the code.

Comment (Jordie0608) [inline SQL/tgstation_schema.sql]: Just to keep it inline with the table used on our server, these columns were all added by errorage to use with his stat-tracking site and he never updated the code with them.
That said, I don't think they have any function beyond record-keeping.

Comment (Markolie) [inline SQL/tgstation_schema.sql]: Makes sense, thanks.

**Comment by Razharas** [comment]
@Jordie0608 
The status of this?

**Comment by Jordie0608** [comment]
It's ready to merge.

**Comment by duncathan** [comment]
It still bothers me that there are no defines here

**Comment by Jordie0608** [comment]
Blame erro

**Comment by duncathan** [comment]
Past shit is no excuse for present shit

**Comment by Razharas** [comment]
>SQL

Didnt even read
Playtest is all it needs

**Pull request merged**